src/compiler/checker.ts

```diff
@@ -17897,6 +17899,9 @@ namespace ts {
                     errorInfo = chainDiagnosticMessages(errorInfo, Diagnostics.Property_0_does_not_exist_on_type_1_Did_you_mean_2, declarationNameToString(propNode), typeToString(containingType), suggestedName);
                     relatedInfo = suggestion.valueDeclaration && createDiagnosticForNode(suggestion.valueDeclaration, Diagnostics._0_is_declared_here, suggestedName);
                 }
+                else if (isStaticPropertyExists(propNode, containingType)) {
+                    errorInfo = chainDiagnosticMessages(errorInfo, Diagnostics.Property_0_is_declared_static_and_can_not_be_called_from_an_instance, declarationNameToString(propNode));
+                }

```

Comment:
maybe this should work on element access too? I don't think it calls reportNonexistentProperty, but similar errors exist in getPropertyTypeForIndexType.

If you do have to copy the code into getPropertyTypeForIndexType, we should think about refactoring it later to call reportNonexistentProperty.

Either way, I'd like to see some tests of element access like `c["bar"]`.

Reply:
> maybe this should work on element access too?
Ok, I'll check, thanks